source/lib.js

```diff
@@ -6,6 +6,50 @@
  * here eventually.
  */
 
+/**
+ * Duplicate a block and attach statements from the clause to be prefixed to the block.
+ * Adds braces if the block is bare.
+ *
+ * @returns the duplicated block with prefix statements attached.
+ */
+function blockWithPrefix(clause, block) {
+  if (!clause) return block
+  const { blockPrefix } = clause
```

Comment:
This API feels a little weird, but I guess it leaves room for passing in content other than `blockPrefix`? It might be cleaner to just pass in `clause?.blockPrefix` though.

From the name "blockWithPrefix", I'd also expect the arguments to be in the other order.